File: Turquoise/UI/TQUserInfoInputView.m

#import "TQUserInfoInputView.h"

@implementation TQUserInfoInputView {
  IBOutlet TQTextField *_userFullNameTextField;
  IBOutlet TQTextField *_userEmailTextField;
}

- (IBAction)proceedButtonDidTap:(id)sender {
  //~TA TODO: fix
//  NSString *userFullName = [_userFullNameTextField.text tq_whitespaceAndNewlineStrippedString];
//  NSString *userEmail = [_userEmailTextField.text tq_whitespaceAndNewlineStrippedString];
  NSString *userFullName = _userFullNameTextField.text;
  NSString *userEmail = _userEmailTextField.text;

  if (userFullName.length > 0 && userEmail.length > 0) {
    // TODO: verify email format
    if (_completionBlock) {
      _completionBlock(userFullName, userEmail);
    }
    [[TQOverlay sharedInstance] dismissWithAnimated:YES];
  }
}

@end

File: Turquoise/UI/TQUserInfoInputView.swift
import Foundation
import UIKit

typealias TQUserInfoInputViewCompletionBlock = (String, String) -> Void

class TQUserInfoInputView : UIView {
  var completionBlock: TQUserInfoInputViewCompletionBlock?

  @IBOutlet var userFullNameTextField: PasswordField!
  @IBOutlet var userEmailTextField: PasswordField!

  @IBAction func proceedButtonDidTap(_ sender: Any?) {
    let userFullName = self.userFullNameTextField.text?.tq_whitespaceAndNewlineStrippedString ?? ""
    let userEmail = self.userEmailTextField.text?.tq_whitespaceAndNewlineStrippedString ?? ""
    guard !userFullName.isEmpty, !userEmail.isEmpty else {
      return
    }

    // TODO: verify email format
    self.completionBlock?(userFullName, userEmail)
    TQOverlay.sharedInstance.dismiss(animated: true)
  }
}
